### PR TITLE
[FIX] core: accounting groups field in user form view

### PR DESCRIPTION
**Pull request by rco-odoo**
This is a followup of https://github.com/odoo/odoo/pull/63423.

The source of the bug is the special case introduced for accounting
groups in d8c5cc1335806d379134866262840fcafb05b643.  The groups in that
selection field are not totally ordered, and the non-ordered elements
are de facto ordered by name, which is a translated field!  In the
example above, the English version uses the field name
`sel_groups_22_23_24_25` while the Dutch version uses the field name
`sel_groups_23_22_24_25`.  Because the first one is used in the form
view, and it is not found in the model's documented fields (which uses
the second one), the field is discarded from the view.

The first fix was modifying method fields_get() to return all possible
field names for that selection field.  But this triggers memory errors
when you duplicate the groups that are partially ordered: having N such
groups generates N! permutations, which does not scale with N.

So, instead of generating all possible field names, we generate the
exact field name used in the user form view, i.e., the one using no
translations for the group names.

### CODE REVIEW

File: odoo/addons/base/models/res_users.py [hunk 1]
```diff
@@ -1450,36 +1450,28 @@ def fields_get(self, allfields=None, attributes=None):
...
-                        'type': 'selection',
-                        'string': app.name or _('Other'),
-                        'selection': selection_vals + [(g.id, g.name) for g in gs],
-                        'help': '\n'.join(tips),
-                        'exportable': False,
-                        'selectable': False,
-                    }
+                    # put field_name_gs in the same order as in the user form view
+                    order = {g: len(g.trans_implied_ids & gs) for g in gs}
+                    field_name_gs = gs.with_context(lang=None).sorted('name').sorted(order.get)
```
Comment (KangOl) [inline odoo/addons/base/models/res_users.py:1460]: Shouldn't it be done for all "app"? I don't see why it needs to be specialized for `base.module_category_accounting_accounting` only.

Comment (rco-odoo) [reply odoo/addons/base/models/res_users.py:1460]: By design we use a selection field only when the corresponding groups are totally ordered.
However, an exception was introduced for the accounting groups: they are not totally ordered but we still use a selection field.  So this is only needed for that category of groups.

**Review by aj-fuentes — APPROVED** [review]
LGTM

Tested on clean DB with only `account_accountant` installed. It's now possible to duplicate `Groups > Accounting > Accountant` more than 8 times (previous cap without this fix). This was the simplest reproducer of the issue fixed here.

upg-12548 successfully finishes now (original source of the issue).

**Comment by rco-odoo** [comment]
@robodoo r+

**Pull request closed without merging**